### PR TITLE
🐛 Source LinkedIn Ads: fixed issue with normalization on redshift

### LINKED ISSUE #13018: Sync failed LinkedIn Ads
## Environment
- **Airbyte version**:  example is 0.35.57-alpha
- **OS Version / Instance**:  AWS EC2
- **Deployment**: Docker
- **Source Connector and version**: LinkedIn Ads
- **Destination Connector and version**: Redshift
- **Severity**: Medium
- **Step where error happened**: Sync job

## Current Behavior
When setting up a connection between LinkedIn ads and Redshift the sync job fails for the ad_campgain_analytics and ad_creative_analytics stream, due to a syntax error. The error logs are attached

## Expected Behavior
It should sync normally I suppose

## Logs
[logs-3895.txt](https://github.com/airbytehq/airbyte/files/8729188/logs-3895.txt)

### LINKED ISSUE #13492: 🎉  Source LinkedIn Ads: Increase `unit_test` coverage up to 90%

### LINKED ISSUE #13493: 🎉  Source LinkedIn Ads: label LinkedIn Ads as `generally_available` in UI

### LINKED ISSUE #13494: Re-factor public documentation to reflect `standard_template`

### PR DESCRIPTION
**Pull request by bazarnov**
## What
Resolving: https://github.com/airbytehq/airbyte/issues/13018

## How
- renamed `pivot` key in Analytics streams, to be `_pivot` to avoid name conflicts on Redshift, since `pivot` is the reserved keyword.
- re-factored the public documentation to reflect `standard_template`
- increased `unit_test` coverage to 92%
- labeled `source-linkedin-ads` as `generally_available` for UI

## 🚨 User Impact 🚨
Customers should `Refresh source schema` for all existing connections, after the update to `0.1.8` since the schema should also reflect the updated key name. Or do not update if your existing connection is working properly and your destination is different from Redshift.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [X] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/utils.py [hunk 1]
```diff
@@ -302,6 +294,20 @@ def transform_variables(
...
+    """
+    Rename records keys (columns) indicated in `dict_keys` to avoid normalization issues for certain destinations.
+    Example:
+        The `pivot` or `PIVOT` is the reserved keyword for DESTINATION REDSHIFT, we should avoid using it in this case.
+        https://github.com/airbytehq/airbyte/issues/13018
+    """
+    for key in dict_keys:
+        if key in record:
+            record[f"_{key}"] = record[key]  # create new key from original
+            record.pop(key)  # remove the original key
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/utils.py:307]:
```suggestion
        record[f"_{key}"] =  record.pop(key) if key in record # create new key from original
```

File: airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/utils.py [hunk 2]
```diff
@@ -321,4 +327,9 @@ def transform_data(records: List) -> Iterable[Mapping]:
         if "variables" in record:
             record = transform_variables(record)
 
+        # replace `pivot` with `_pivot`, to allow redshift normalization,
+        # since `pivot` is a reserved keyword for Redshift.
+        # on behalf of https://github.com/airbytehq/airbyte/issues/13018
+        record = transform_col_names(record, ["pivot"])
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/utils.py:333]: It's look like `["pivot"]` can be as variable (e.g. `destination reserved names`)

**Review by lazebnyi — APPROVED** [review]
LGTM. Few small comments.

File: airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/utils.py (hunk 1 above)
Comment (bazarnov) [reply airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/utils.py:307]: We have to assign the old value first, then remove, this suggestion doesn't cover the assignment part. Leaving as it was.

File: airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/utils.py (hunk 2 above)
Comment (bazarnov) [reply airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/utils.py:333]: Done, good catch.

**Pull request merged**